staging/src/k8s.io/apiserver/pkg/storage/etcd/etcd_helper.go

```diff
@@ -347,6 +347,7 @@ func (h *etcdHelper) GetToList(ctx context.Context, key string, resourceVersion
 	metrics.RecordEtcdRequestLatency("get", getTypeName(listPtr), startTime)
 	if err != nil {
 		if etcdutil.IsEtcdNotFound(err) {
+			// TODO: ResourceVersion should be set here.
```

Comment:
Maybe it's a test issue, but in the test that I added, I'm getting "response=nil" in this case...

Reply:
Actually - this seem to be an issue of etcd client library:
https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/coreos/etcd/client/keys.go#L654

In case of status different than OK and Created (so 200 and 201), the returned response is nil.

Reply:
Aaah ok - thanks. It works now. PTAL